warehouse/integrations/snowflake/snowflake.go

```diff
@@ -143,6 +145,15 @@ type optionalCreds struct {
 	schemaName string
 }
 
+type duplicateMessage struct {
+	id         string
+	receivedAt time.Time
+}
+
+func (m *duplicateMessage) String() string {
+	return fmt.Sprintf("%s@%s", m.id, m.receivedAt.Format("2006-01-02 15:04:05"))
```

Comment:
[optional] This can be called for lots of messages in a loop and we're talking a simple string interpolation. I suggest not using reflection here.

Reply:
Good point. Simple interpolation would work here. Just for logging purposes.